datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -1656,26 +1655,50 @@ mod tests {
             object_meta: meta,
             partition_values: vec![
                 ScalarValue::Utf8(Some("2021".to_owned())),
-                ScalarValue::Utf8(Some("10".to_owned())),
-                ScalarValue::Utf8(Some("26".to_owned())),
+                ScalarValue::UInt8(Some(10)),
+                ScalarValue::Dictionary(
+                    Box::new(DataType::UInt16),
+                    Box::new(ScalarValue::Utf8(Some("26".to_owned()))),
+                ),
             ],
             range: None,
             extensions: None,
         };
 
+        let expected_schema = Schema::new(vec![
+            Field::new("id", DataType::Int32, true),
+            Field::new("bool_col", DataType::Boolean, true),
+            Field::new("tinyint_col", DataType::Int32, true),
```

Comment:
thats looks like typo -> `tinyint_col` should be `DataType::UInt8` ?

Reply:
The schema that I get from `infer_schema` (see two statements above) contains an `Int32`. I haven't touched this code and this is the current state on `main`. Not saying this is correct though, it might be a pre-existing bug. Note that the schema wasn't tested before, that's why I added this assertion.